build_deploy.sh

```diff
@@ -2,10 +2,12 @@
 
 set -exv
 
-source 'deploy/build-deploy-common.sh'
+source 'scripts/deploy/build-deploy-common.sh'
 
 BACKWARDS_COMPATIBILITY_TAGS="latest qa"
 IMAGE_NAME="${IMAGE_NAME:-quay.io/cloudservices/compliance-backend}"
-IMAGE_TAG="${IMAGE_TAG}"
+IMAGE_TAG="${IMAGE_TAG:?}"
+
+git submodule update --init
```

Comment:
Doesn't this need `--remote`?
From my own experience, git submodules are not that easy to work with. The value that is checked out is stored and kept, and usually it should be committed. Please check that this would pull new content from the remote and move the HEAD, esp. when this will be run on shared (test) environment. 

Configuration lacks branch tracking.

Reply:
If the intention is to have this only run on a clean pull and never have it updated after that, then this would work well. The problematic part would be when one would have existing repository pulled (i.e. wants to test the script), then the behavior of this might not be correct (at least not w/o `--remote`).

The ref of the submodule can be committed to the repository. That would pin it to specific commit. Then on pull (for test runs) there is an option to pull submodules as well. Configuration might need a tracking branch, just for the ease of updates.  ... What I found as the easiest way of updating the submodule, is to `cd` to its folder do a `pull` and then go back and commit the `ref` to the main repository.

Let's first clear out what do we want to achieve with this.

Reply:
I'd suggest to try and test this deploy script with all cases in mind: having a clean repo, having the submodule already checked, having the submodule checked but with pending changes that needs to be pulled.

Reply:
@vkrizan  tested:

- having a clean repo, no upstream changes: on a fresh repo clone (which is what happens on CI every time) - the `build_deploy` runs the line in build_deploy that does the submodule init, doing a checkout of the commit under version control that was added originally.

- having a clean repo, upstream changes: the SHA of the committed version of the submodule is the one used in the checkout by git on the init operation (as expected). Therefore, the latest changes won't be checked out . 

- having an existing repo with the submodule initialized, upstream changes: the SHA of the committed version of the submodule is the one used in the checkout by git on the init operation (again, as expected). if we want to pull the latest changes from upstream, we must add the `--remote` flag , and *add and commit* those changes (hence, proactively pulling the latest changes. I think this is expected, we may not want the "latest updates always", in the same way we have pinned versions for the other libraries we use.

@vkrizan  please let me know what's required to get this merged. we can meet if needed

Reply:
@skateman thoughts on this approach? Technically it is a better solution that what we have today, but personally I'm bit hesitant as it might be harder to maintain (as git submodules are a bit advanced topic). I think that it could be brought up in a wider discussion with tenants and platform. Although, I guess it should not block this PR.